### ISSUE TITLE
Autocompletion is failing very often

### ISSUE THREAD

**Issue by babobski** [Komodo/KomodoEdit#3601]
### Short Summary
When working with php/html/javascript files the autocompletion is failing most of the time.
It feels very laggy and sometimes I get auto completions, but most of the time it does not.
If I look to my log file it looks like the codeintell is rebooting a lot.

### Platform Information
```
[2018-09-06 09:39:23,190] [INFO] Startup: Welcome to Komodo IDE 11.1.1 build 91042 (platform win32-x86, running on Windows post2012Server version 6.3.9600)
[2018-09-06 09:39:23,190] [INFO] Startup: C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\mozilla\komodo.exe built on Tue Aug 28 01:01:20 2018
[2018-09-06 09:39:24,035] [WARNING] console-logger: mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create (1) in resource://gre/modules/Preferences.jsm:381
[2018-09-06 09:39:24,098] [WARNING] root: [object Object].logging has been converted to a CommonJS module; use require("ko/logging") instead (since Komodo 9.0.0a1).
    @chrome://komodo/content/komodo.js:15:1

[2018-09-06 09:39:24,410] [ERROR] console-logger: bad script XDR magic number (0) in :0
Traceback (most recent call last):
  File "", line 0, in 

[2018-09-06 09:39:25,676] [WARNING] console-logger: Use of Mutation Events is deprecated. Use MutationObserver instead. (1) in chrome://komodo/content/bindings/scrolltabs.xml:1184
[2018-09-06 09:39:26,723] [WARNING] ko.widgets: Pref path doesn't exist: windowWorkspace, 1
[2018-09-06 09:39:26,785] [WARNING] ko.widgets: Could not restore widget unittest-widget
[2018-09-06 09:39:26,832] [WARNING] ko.widgets: Could not restore widget symbolbrowser-widget
[2018-09-06 09:39:26,973] [DEBUG] ko.launch: versioncheck on http://docs.komodoide.com/changelog/11
[2018-09-06 09:39:28,019] [INFO] koInitService: Adding pre startup service for 'koFileStatusService': '@activestate.com/koFileStatusService;1'
[2018-09-06 09:39:28,128] [INFO] komodospellchecker: Spell checker loaded.
[2018-09-06 09:39:28,128] [INFO] komodospellchecker: Using dictionary en-US
[2018-09-06 09:39:28,207] [INFO] elastic_tabstops: Elastic tabstops loaded.
[2018-09-06 09:39:28,285] [INFO] codeintel/process: Starting CodeIntel
[2018-09-06 09:39:28,285] [DEBUG] codeintel/process: PYTHONPATH: C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\python\..\lib\python2.7;C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\support\dbgp\pythonlib;C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\mozilla\extensions\codeintel@activestate.com\content\..\pylib\codeintel\lib;C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\mozilla\extensions\codeintel@activestate.com\content\..\pylib\codeintel\env\Lib\site-packages
[2018-09-06 09:39:29,019] [ERROR] console-logger: TypeError: this.tabbox.selectedPanel is null (2) in chrome://komodo/content/bindings/scrolltabs.xml:1269
Traceback (most recent call last):
  File "chrome://komodo/content/bindings/scrolltabs.xml", line 1269, in 

[2018-09-06 09:39:29,112] [ERROR] console-logger: TypeError: this.tabbox.selectedPanel is null (2) in chrome://komodo/content/bindings/scrolltabs.xml:1269
Traceback (most recent call last):
  File "chrome://komodo/content/bindings/scrolltabs.xml", line 1269, in 
[2018-09-06 09:39:32,926] [INFO] koInitService: Adding pre startup service for 'KoMemoryReporter': '@activestate.com/koMemoryReporter;1'
[2018-09-06 09:39:32,926] [INFO] koInitService: Adding pre startup service for 'koCommandmentService': '@activestate.com/koCommandmentService;1'
[2018-09-06 09:39:32,926] [INFO] koInitService: Adding pre startup service for 'koDBGPManager': '@activestate.com/koDBGPManager;1'
[2018-09-06 09:39:33,035] [WARNING] console-logger: XUL box for tab element contained an inline #text child, forcing all its children to be wrapped in a block. (1) in file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/koLessProtocolHandler.js:154
[2018-09-06 09:39:33,035] [WARNING] console-logger: XUL box for tab element contained an inline #text child, forcing all its children to be wrapped in a block. (1) in file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/koLessProtocolHandler.js:154
[2018-09-06 09:39:33,035] [WARNING] console-logger: XUL box for tab element contained an inline #text child, forcing all its children to be wrapped in a block. (1) in file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/koLessProtocolHandler.js:154
[2018-09-06 09:39:33,035] [WARNING] console-logger: XUL box for tab element contained an inline #text child, forcing all its children to be wrapped in a block. (1) in file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/koLessProtocolHandler.js:154
[2018-09-06 09:39:33,035] [WARNING] console-logger: XUL box for tab element contained an inline #text child, forcing all its children to be wrapped in a block. (1) in file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/koLessProtocolHandler.js:154
[2018-09-06 09:39:33,035] [WARNING] console-logger: XUL box for tab element contained an inline #text child, forcing all its children to be wrapped in a block. (1) in file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/koLessProtocolHandler.js:154
[2018-09-06 09:39:33,035] [WARNING] console-logger: mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create (1) in file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/koamAddonManager.js:138
[2018-09-06 09:39:33,519] [WARNING] console-logger: XUL box for tab element contained an inline #text child, forcing all its children to be wrapped in a block. (1) in chrome://unittest/content/views/panel.xul:0
[2018-09-06 09:39:33,519] [WARNING] console-logger: XUL box for tab element contained an inline #text child, forcing all its children to be wrapped in a block. (1) in chrome://unittest/content/views/panel.xul:0
[2018-09-06 09:39:33,519] [WARNING] console-logger: XUL box for tab element contained an inline #text child, forcing all its children to be wrapped in a block. (1) in chrome://unittest/content/views/panel.xul:0
[2018-09-06 09:39:34,676] [DEBUG] codeintel/process: stdout: 
port:54903

pid:520

[2018-09-06 09:39:34,676] [INFO] codeintel/process: CodeIntel started on port 54903
[2018-09-06 09:39:34,690] [DEBUG] codeintel/process: Calling callbacks for: started, number: 1
[2018-09-06 09:39:36,035] [DEBUG] ko.launch: Response: http://docs.komodoide.com/__omnigollum__/auth/github?origin=%2Fcreate%2Fchangelog%2F11 (404)
[2018-09-06 09:39:36,035] [DEBUG] ko.launch: versioncheck on http://docs.komodoide.com/changelog/111
[2018-09-06 09:39:36,769] [DEBUG] ko.launch: Response: http://docs.komodoide.com/__omnigollum__/auth/github?origin=%2Fcreate%2Fchangelog%2F111 (404)
[2018-09-06 10:03:31,131] [WARNING] console-logger: Key event not available on GTK2: key="u" modifiers="control,shift" (1) in chrome://komodo/content/komodo.xul:0
[2018-09-06 10:04:05,865] [ERROR] codeintel/feature/calltips: Failed updating calltip panel
-- EXCEPTION START --
TypeError: activePanel.update is not a function
+ stack
    onReceivedCompletions@chrome://codeintel/content/sdk/feature/calltips.js:161:17
-- EXCEPTION END --
[2018-09-06 10:51:22,842] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute '__getitem__' (-32603)
[2018-09-06 10:51:22,900] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute '__getitem__' (-32603)
[2018-09-06 10:51:24,173] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute '__getitem__' (-32603)
[2018-09-06 10:51:24,234] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute '__getitem__' (-32603)
[2018-09-06 10:51:24,371] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute '__getitem__' (-32603)
[2018-09-06 10:51:25,759] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute '__getitem__' (-32603)
[2018-09-06 10:51:52,288] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:52,428] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:52,536] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:52,742] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:53,108] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:53,269] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:53,434] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:53,770] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:54,020] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:54,068] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:54,197] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:54,467] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:54,970] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:55,085] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:55,240] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:55,355] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:55,526] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)
[2018-09-06 10:51:55,790] [ERROR] codeintel/service/mediator/codeintel: Request failed with message: Failed calling API method: getCompletions, message: 'NoneType' object has no attribute 'define' (-32603)

[2018-09-06 10:52:55,533] [ERROR] socket-tcp: Error occurred: error : NetworkError
Traceback from ERROR in 'socket-tcp' logger:
    Logger.prototype.error@chrome://komodo/content/sdk/logging.js:411:27
    connection/onError@chrome://komodo/content/sdk/socket/tcp.js:101:17
    ts_callListener@file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/TCPSocket.js:340:5
    TCPSocket.prototype._maybeReportErrorAndCloseIfOpen@file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/TCPSocket.js:828:7
    ts_onStopRequest@file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/TCPSocket.js:885:5

[2018-09-06 10:52:55,637] [INFO] codeintel/process: onClose called, code: 9
[2018-09-06 10:52:55,638] [INFO] codeintel/process: Process stopped without a request to stop it, restarting ..
[2018-09-06 10:52:55,638] [INFO] codeintel/process: Restarting CodeIntel
[2018-09-06 10:52:55,638] [INFO] codeintel/process: Starting CodeIntel
[2018-09-06 10:52:55,641] [DEBUG] codeintel/process: PYTHONPATH: C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\python\..\lib\python2.7;C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\support\dbgp\pythonlib;C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\mozilla\extensions\codeintel@activestate.com\content\..\pylib\codeintel\lib;C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\mozilla\extensions\codeintel@activestate.com\content\..\pylib\codeintel\env\Lib\site-packages
[2018-09-06 10:52:56,598] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:52:58,414] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:52:59,312] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:01,496] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:01,888] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:01,990] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:02,105] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:02,309] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:02,430] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:02,532] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:03,022] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:03,188] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:03,691] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:03,846] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:04,068] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:04,177] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:04,308] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:04,572] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:04,891] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:05,010] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:05,210] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:05,323] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:05,426] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:05,667] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:05,859] [WARNING] codeintel/service/mediator: Waiting for mediator
[2018-09-06 10:53:06,069] [DEBUG] codeintel/process: stdout: 
port:55411

pid:12840

[2018-09-06 10:53:06,069] [INFO] codeintel/process: CodeIntel started on port 55411
[2018-09-06 10:53:06,069] [DEBUG] codeintel/process: Calling callbacks for: started, number: 1
[2018-09-06 10:54:16,165] [ERROR] socket-tcp: Error occurred: error : NetworkError
Traceback from ERROR in 'socket-tcp' logger:
    Logger.prototype.error@chrome://komodo/content/sdk/logging.js:411:27
    connection/onError@chrome://komodo/content/sdk/socket/tcp.js:101:17
    ts_callListener@file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/TCPSocket.js:340:5
    TCPSocket.prototype._maybeReportErrorAndCloseIfOpen@file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/TCPSocket.js:828:7
    ts_onStopRequest@file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/TCPSocket.js:885:5

[2018-09-06 10:54:16,279] [INFO] codeintel/process: onClose called, code: 9
[2018-09-06 10:54:16,279] [INFO] codeintel/process: Process stopped without a request to stop it, restarting ..
[2018-09-06 10:54:16,279] [INFO] codeintel/process: Restarting CodeIntel
[2018-09-06 10:54:16,279] [INFO] codeintel/process: Starting CodeIntel
[2018-09-06 10:54:16,283] [DEBUG] codeintel/process: PYTHONPATH: C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\python\..\lib\python2.7;C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\support\dbgp\pythonlib;C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\mozilla\extensions\codeintel@activestate.com\content\..\pylib\codeintel\lib;C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\mozilla\extensions\codeintel@activestate.com\content\..\pylib\codeintel\env\Lib\site-packages
[2018-09-06 10:54:21,950] [DEBUG] codeintel/process: stdout: 
port:55429

pid:13036

[2018-09-06 10:54:21,950] [INFO] codeintel/process: CodeIntel started on port 55429
[2018-09-06 10:54:21,950] [DEBUG] codeintel/process: Calling callbacks for: started, number: 1
[2018-09-06 10:55:46,976] [ERROR] socket-tcp: Error occurred: error : NetworkError
Traceback from ERROR in 'socket-tcp' logger:
    Logger.prototype.error@chrome://komodo/content/sdk/logging.js:411:27
    connection/onError@chrome://komodo/content/sdk/socket/tcp.js:101:17
    ts_callListener@file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/TCPSocket.js:340:5
    TCPSocket.prototype._maybeReportErrorAndCloseIfOpen@file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/TCPSocket.js:828:7
    ts_onStopRequest@file:///C:/Program%20Files%20(x86)/ActiveState%20Komodo%20IDE%2011%20nightly/lib/mozilla/components/TCPSocket.js:885:5

[2018-09-06 10:55:47,055] [INFO] codeintel/process: onClose called, code: 9
[2018-09-06 10:55:47,055] [INFO] codeintel/process: Process stopped without a request to stop it, restarting ..
[2018-09-06 10:55:47,055] [INFO] codeintel/process: Restarting CodeIntel
[2018-09-06 10:55:47,055] [INFO] codeintel/process: Starting CodeIntel
[2018-09-06 10:55:47,059] [DEBUG] codeintel/process: PYTHONPATH: C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\python\..\lib\python2.7;C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\support\dbgp\pythonlib;C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\mozilla\extensions\codeintel@activestate.com\content\..\pylib\codeintel\lib;C:\Program Files (x86)\ActiveState Komodo IDE 11 nightly\lib\mozilla\extensions\codeintel@activestate.com\content\..\pylib\codeintel\env\Lib\site-packages
[2018-09-06 10:55:52,782] [DEBUG] codeintel/process: stdout: 
port:55451

pid:1224

[2018-09-06 10:55:52,782] [INFO] codeintel/process: CodeIntel started on port 55451
[2018-09-06 10:55:52,782] [DEBUG] codeintel/process: Calling callbacks for: started, number: 1
[2018-09-06 10:58:12,346] [ERROR] console-logger: TypeError: n is undefined (2) in chrome://komodo/content/contrib/commonjs/underscore.js:5
Traceback (most recent call last):
  File "chrome://komodo/content/contrib/commonjs/underscore.js", line 5, in 

[2018-09-06 11:03:22,552] [ERROR] console-logger: TypeError: pos is undefined (2) in chrome://komodo/content/sdk/editor.js:1300
Traceback (most recent call last):
  File "chrome://komodo/content/sdk/editor.js", line 1300, in 

[2018-09-06 11:06:16,859] [ERROR] console-logger: TypeError: pos is undefined (2) in chrome://komodo/content/sdk/editor.js:1300
Traceback (most recent call last):
  File "chrome://komodo/content/sdk/editor.js", line 1300, in 

[2018-09-06 11:06:16,914] [ERROR] console-logger: TypeError: pos is undefined (2) in chrome://komodo/content/sdk/editor.js:1300
Traceback (most recent call last):
  File "chrome://komodo/content/sdk/editor.js", line 1300, in 

[2018-09-06 11:09:59,687] [WARNING] console-logger: Key event not available on GTK2: key="u" modifiers="control,shift" (1) in chrome://komodo/content/tail/tail.xul:0

```

**Comment by babobski** [comment]
Seems to be fixed in V12

**Comment by goozak** [comment]
Just for those who finds this later on... I had so much issues with autocompletion when v11 was released that I stayed on v10 (which, to be honest, also had issues).  I recently discovered that those issues where likely coming from the old project file I was using, since v8 or v9...  I recently had to completely reinstall my work computer and started fresh with v11.1 and a new project file — everything is working fine!  Autocompletion and Go to anything both work fine.
